File: app/src/main/java/com/ruideraj/secretelephant/injection/ContactsModule.java
package com.ruideraj.secretelephant.injection;

import android.app.Application;

import com.ruideraj.secretelephant.contacts.ContactsDao;
import com.ruideraj.secretelephant.contacts.ContactsRepository;

import dagger.Module;
import dagger.Provides;

@Module
public class ContactsModule {

    @Provides
    ContactsRepository providesContactsRepository(ContactsDao dao) {
        return new ContactsRepository(dao);
    }

    @Provides
    ContactsDao providesContactsDao(Application application) {
        return new ContactsDao(application);
    }
}

File: app/src/main/java/com/ruideraj/secretelephant/injection/ContactsModule.kt
package com.ruideraj.secretelephant.injection

import com.ruideraj.secretelephant.contacts.ContactsDao
import com.ruideraj.secretelephant.contacts.ContactsProvider
import com.ruideraj.secretelephant.contacts.ContactsRepository
import com.ruideraj.secretelephant.contacts.ContactsRepositoryImpl
import dagger.Module
import dagger.Provides

@Module
class ContactsModule {

    @Provides
    fun providesContactsRepository(contactsRepoImpl: ContactsRepositoryImpl): ContactsRepository
            = contactsRepoImpl

    @Provides
    fun providesContactsProvider(contactsDao: ContactsDao): ContactsProvider = contactsDao

}